sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala

```diff
@@ -71,6 +71,18 @@ private[execution] sealed trait HashedRelation extends KnownSizeEstimation {
    */
   def keyIsUnique: Boolean
 
+  /**
+   * Note that, the hashed relation can be empty despite the Iterator[InternalRow] being not empty,
+   * since the hashed relation skips over null keys.
+   */
```

Comment:
Reading this doc, sounds like `inputEmpty` returns true if there are null keys. But I saw

```
def inputEmpty: Boolean = numKeys == 0 && !anyNullKeyExists
``` 

Can you rephase this doc to clearly state when `inputEmpty` returns true?

Reply:
during building a hashedRelation, it could end up following case.

1. input: iterator[InternalRow] itself is empty, than inputEmpty = true
2. iterator contains row that has null values in any column, anyNullKeyExists = true, but row got filtered and not present in hashedRelation.
3. normal not null row will be kept.

inputEmpty != !anyNullKeyExists
because even no null key row does not exist, normal row could still be. 

this should be right, but i will add some comments to make it clear.
def inputEmpty: Boolean = numKeys == 0 && !anyNullKeyExists